cipd_packages/device_doctor/lib/src/android_device.dart

```diff
@@ -453,14 +453,14 @@ class AndroidDevice implements Device {
     processManager ??= LocalProcessManager();
     final int timeoutSecs = 60;
     print('Device recovery: deleting package caches...');
-    await eval('adb', <String>['wait-for-device'], canFail: false, processManager: processManager)
+    await eval('adb', <String>['wait-for-device', 'shell', 'while [[ -z \$(getprop sys.boot_completed) ]]; do sleep 1; done;'], canFail: false, processManager: processManager)
```

Comment:
FYI: this is what I was referring to when I tried this through similar code. Are you able to run this command through the ProcessManager and have it succeed without error?

Reply:
See this: https://discord.com/channels/608014603317936148/846507907876257822/1156718853921964052

Reply:
What is the easier way you found?

Reply:
I just did not include the loop as I could not get it to run.

Reply:
The wait-for-device should at least be able to detect the attached device.